### PR TITLE
Add SideBar width control by theme. 

### PR DESCRIPTION
**Pull request by whyuan1976**
Like this : 

const myTheme = {
   .......
   sideBar : {
        width:'16em',
    },
};

### CODE REVIEW

File: src/mui/layout/Sidebar.js [hunk 1]
```diff
@@ -6,29 +6,51 @@ import Paper from 'material-ui/Paper';
 import Responsive from './Responsive';
 import { setSidebarVisibility as setSidebarVisibilityAction } from '../../actions';
 
-const styles = {
+
+export function getStyles(props, context) {
+  const {
+    sideBar,
+  } = context.muiTheme;
+
```
Comment (djhi) [inline src/mui/layout/Sidebar.js]: Please remove all the extra lines (just keep one)

File: src/mui/layout/Sidebar.js [hunk 2]
```diff
@@ -6,29 +6,51 @@ import Paper from 'material-ui/Paper';
...
+  
+
+  const styles = {
     sidebarOpen: {
-        flex: '0 0 16em',
+        flex: !(sideBar&&sideBar.width)?'0 0 16em':`0 0 ${sideBar.width}`,
```
Comment (djhi) [inline src/mui/layout/Sidebar.js]: add spaces around `&&`, `?` and `:`

File: src/mui/layout/Sidebar.js [hunk 3]
```diff
@@ -6,29 +6,51 @@ import Paper from 'material-ui/Paper';
...
         marginLeft: 0,
         order: -1,
         transition: 'margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
     },
     sidebarClosed: {
-        flex: '0 0 16em',
-        marginLeft: '-16em',
+        flex: !(sideBar&&sideBar.width)?'0 0 16em':`0 0 ${sideBar.width}`,
```
Comment (djhi) [inline src/mui/layout/Sidebar.js]: same

File: src/mui/layout/Sidebar.js [hunk 4]
```diff
@@ -6,29 +6,51 @@ import Paper from 'material-ui/Paper';
...
+        marginLeft: !(sideBar&&sideBar.width)?'-16em':`-${sideBar.width}`,
```
Comment (djhi) [inline src/mui/layout/Sidebar.js]: same

File: src/mui/layout/Sidebar.js [hunk 5]
```diff
@@ -6,29 +6,51 @@ import Paper from 'material-ui/Paper';
...
         order: -1,
         transition: 'margin 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms',
     },
-};
+  };
+
+  return styles;
+}
+
+
```
Comment (djhi) [inline src/mui/layout/Sidebar.js]: Too many extra lines here too

File: src/mui/layout/Sidebar.js [hunk 6]
```diff
@@ -6,29 +6,51 @@ import Paper from 'material-ui/Paper';
...
+
+
 
 // We shouldn't need PureComponent here as it's connected
 // but for some reason it keeps rendering even though mapStateToProps returns the same object
 class Sidebar extends PureComponent {
+
```
Comment (djhi) [inline src/mui/layout/Sidebar.js]: extra line

File: src/mui/layout/Sidebar.js [hunk 7]
```diff
@@ -6,29 +6,43 @@ import Paper from 'material-ui/Paper';
...
-const styles = {
+
+export function getStyles(props, context) {
+  const {
+    sideBar,
+  } = context.muiTheme;  
+  const styles = {
     sidebarOpen: {
-        flex: '0 0 16em',
+        flex: !(sideBar && sideBar.width) ? '0 0 16em' : `0 0 ${sideBar.width}`,
```
Comment (fzaninotto) [inline src/mui/layout/Sidebar.js]: please don't negate the condition, it makes it harder to reason about. Same for the others.

File: src/mui/layout/Sidebar.js [hunk 8]
```diff
@@ -6,29 +6,43 @@ import Paper from 'material-ui/Paper';
...
+  };
+
+  return styles;
+}
 
 // We shouldn't need PureComponent here as it's connected
 // but for some reason it keeps rendering even though mapStateToProps returns the same object
 class Sidebar extends PureComponent {
+
+    static contextTypes = {
```
Comment (fzaninotto) [inline src/mui/layout/Sidebar.js]: Please use the `muiThemeable` HOC (just like `mui/layout/AppBar.js`), I prefer it over grabbing the theme from the context.

**Comment by djhi** [comment]
Superceded by #651

**Pull request closed without merging**